sasl/sasl.go

```diff
@@ -2,6 +2,8 @@ package sasl
 
 import "context"
 
+type ctxKey struct{}
```

Comment:
How about renaming to be more explicit what value this key is intended to map to?
```suggestion
type metadataKey struct{}
```

Reply:
I'm fine with renaming this, but while `ctxKey` might sound obscure, `metadataKey` is a little bit ambiguous imho. `ctxKey` makes it clear it's a type used to hold stuff into the context. What do you think?

Reply:
I guess I was thinking it may be a bit generic, and not leave a lot of room for adding more keys for other values we'd keep in the context.

It's unexported so easy to change, no real concerns about leaving it as-is for now, it was just a suggestion.